### DISCUSSION TITLE
Are Array/Slice of structs supported?

### DISCUSSION THREAD

**Discussion by csg33k** [earthboundkid/requests#62] [Support] [answered]
I was trying to see if I can return a list of structs instead of just simple struct?   For example:

```go
func TestFoobar(t *testing.T) {
	data := test_common.ReadLocalFile(t, "dummy.json")
	var post []models.Post
	err := requests.
		URL("http://localhost:2368").
		Pathf("/ghost/api/content/posts/").
		ToJSON(&post).
		Param("key", "secret").
		Fetch(context.Background())
	assert.Nil(t, err)
}
```

The other suggestion I had that would be interesting.  It'd be nice nice to support gjson or any other "path" related libraries. Where you can add a path to pull the data from rather than simply starting at the root of the payload.  

For example.  

JSONPath("results.orders")   Any thoughts? It would make this library a lot more powerful IMO.

**Comment by earthboundkid** [comment]
Yes, that works. 

If you want to use a different library for JSON decoding, you can use the ToWriter method and plug it in there. This library only depends on the standard library and /x/net, so it can’t add an alternative JSON parser.

**Comment by csg33k** [comment, marked as answer]
Thanks, that looks great.  I think the combination of those two will do everything I needed.  Much appreciated.
